fvtest/tril/tril/parser.cpp

```diff
@@ -0,0 +1,560 @@
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "ast.h"
+#include "ast.hpp"
+#include <stdio.h>
+#include <string>
+#include <iostream>
+#include <vector>
+#include <iterator>
+#include <sstream>
+
+class Token {
+    public:
+        enum Type {
+            ID = 0,
+            STRING_,
+            STRING_0,
+            INT,
+            WHITESPACE,
+            AT,
+            ZERO,
+            HEXINT,
+            QUOT,
+            LPAREN,
+            RPAREN,
+            LPAREN_SQ,
+            RPAREN_SQ,
+            EQUALS,
+            COMMA,
+            COMMENT,
+            DOUBLE_
+    };
+
+    private:
+        Type type;
+        std::string value;
+    
+
+    public:
+        Token(Type type, std::string value):
+	        type(type), value(value) {}
+        Type getType() const { return type; }
+        void setType(Token::Type t) {type = t ;}
+        const std::string &getValue() const {return value ;}
+};
+
+class LexerFailure {
+	std::string message;
+
+public:
+	LexerFailure(std::string message);
+	const std::string &what() const;
+};
+
+LexerFailure::LexerFailure(std::string message) :
+	message(message) {}
+
+const std::string &LexerFailure::what() const { return message; }
+
+class ParserFailure {
+	std::string message;
+
+public:
+	ParserFailure(std::string message);
+	const std::string &what() const;
+};
+
+ParserFailure::ParserFailure(std::string message) :
+	message(message) {}
+
+const std::string &ParserFailure::what() const { return message; }
+
+bool isNotNewLine(char c) { return c != '\n'; }
+
+bool isNotQuot(char c) { return c != '"'; }
+
+bool isNonZeroDigit(char c) { return '1' <= c && c <= '9'; }
+
+class ASTstates {
+    public:
+        enum State {
+            ID = 0,
+            STRING_,
+            STRING_0,
+            INT,
+            WHITESPACE,
+            AT,
+            ZERO,
+            HEXINT,
+            QUOT,
+            LPAREN,
+            RPAREN,
+            LPAREN_SQ,
+            RPAREN_SQ,
+            EQUALS,
+            COMMA,
+            DOUBLE_,
+
+            START,
+            FAIL,
+            DOT,
+            MINUS,
+            COMMENT,
+            LARGEST_STATE = COMMENT
+        };
+    
+    private:
+        std::vector <State> acceptingStates;
```

Comment:
Please remove the space the `std::vector` and the template arguments (here and in other places):
```suggestion
        std::vector<State> acceptingStates;
```